core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -476,22 +478,33 @@ object Parser extends ParserInstances {
             (left, right) match {
               case (Str(_, s1), Str(_, s2)) => s1.compare(s2)
               case (Str(_, _), _) => -1
-              case (InRange(_, _, _), Str(_, _)) => 1
+              case (OneStr(_, _), Str(_, _)) => 1
+              case (OneStr(_, s1), OneStr(_, s2)) =>
+                val s = s1.lengthCompare(s2.size)
+                if (s == 0) s1.compare(s2)
```

Comment:
why not just `s1.compare(s2)` here?

Reply:
Oops some leftovers from a previous attempt. I will change this.